esp/logging/loggingmanager/loggingmanager.cpp

```diff
@@ -106,6 +106,29 @@ bool CLoggingManager::updateLog(IEspContext& espContext, const char* option, con
     return bRet;
 }
 
+bool CLoggingManager::updateLog(const char* option, IPropertyTree* logInfo, IInterface* extraLog, StringBuffer& status)
```

Comment:
The overloaded interface methods get very complicated very quickly.  Seems like a two part calling model would be better:
creating the log object, setting whatever available parms, and then calling a single updateLog method.
```
IEspLogEntry *entry = logmgr->createLogEntry();
entry->setEspContext(ctx);
entry->setOption(option);
entry->setContent(content);
logmgr->updateLog(entry, status);
```
I might be missing something, but even if we can't change the interface, couldn't these functions be commoned up at least internally?

For example something like:

```
bool updateLogImpl(IEspContext* espContext, const char* option, const char* logContent, StringBuffer& status)
{
  .... real functionality...
}
  
bool updateLog(IEspContext& espContext, const char* option, const char* logContent, StringBuffer& status)
{
    return updateLogImpl(&espContext, option, logContent, status);
}

bool updateLog(IEspContext& espContext, const char* option, const char* logContent, StringBuffer& status)
{
    return updateLogImpl(NULL, option, logContent, status);
}
```